Formula/t/toipe.rb

```diff
@@ -0,0 +1,18 @@
+class Toipe < Formula
+  desc "Yet another typing test, but crab flavoured"
+  homepage "https://github.com/Samyak2/toipe"
+  url "https://github.com/Samyak2/toipe/archive/refs/tags/v0.5.0.tar.gz"
+  sha256 "9916e8fcd4fdbfa38c5fcb70b12fd086b38320c82bbff478be581ae5c1577c8a"
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    version = /toipe 0\.5\.0/
+    assert_match version, shell_output("#{bin}/toipe --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.